src/dlls/mscoree/mscoree.cpp

```diff
@@ -993,6 +896,70 @@ STDAPI GetCORSystemDirectoryInternal(__out_ecount_part_opt(cchBuffer, *pdwLength
     }
     return hr;
 
+#endif // CROSSGEN_COMPILE
+}
+#endif // !FEATURE_CORECLR 
+
+STDAPI GetCORSystemDirectoryInternaL(SString& pBuffer)
```

Comment:
Yes, because of the declaration as an extern "c" function name overloading is not possible.
 The function further takes a global g_hThisInst. 
Any way the other function with original name will be removed as soon as we change the instances in desktop